Can't compile ponyc from source on Arch Linux

I was originally trying to use the Arch Linux packaged version of ponyc but ran into issues as detailed in #2115. I'm falling back to compiling from source, but having issues there too:

```
 make
genopt.cc
genreference.c
gencontrol.c
genfun.c
genlib.c
genexpr.c
gendebug.cc
genobj.c
names.c
syntax.c
finalisers.c
expr.c
flatten.c
import.c
pass.c
serialisers.c
verify.c
docgen.c
sugar.c
scope.c
traits.c
refer.c
casemethod.c
call.c
control.c
fun.c
paths.c
reify.c
compattype.c
subtype.c
sanitise.c
cap.c
lookup.c
alias.c
viewpoint.c
typeparam.c
src/libponyc/type/typeparam.c: In function ‘typeparam_current_inner’:
src/libponyc/type/typeparam.c:606:5: error: this statement may fall through [-Werror=implicit-fallthrough=]
     {
     ^
src/libponyc/type/typeparam.c:613:5: note: here
     default:
     ^~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:700: build/release/obj/libponyc/type/typeparam.o] Error 1
```

I've downgraded my versions of clang and llvm to 3.9.1 (since current Arch versions are 4.*):

```
clang --version
clang version 3.9.1 (tags/RELEASE_391/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

llvm-ar --version
LLVM (http://llvm.org/):
  LLVM version 3.9.1
  Optimized build.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: ivybridge
```

I'm not experienced with Make, but I couldn't find anything in the ponyc Makefile that adds `-Werror=implicit-fallthrough=`. Maybe it's a default on my machine, or on Arch? Can anyone advise on what I should check?

--- a/src/libponyc/type/typeparam.c
+++ b/src/libponyc/type/typeparam.c
@@ -608,6 +608,8 @@ static void typeparam_current_inner(ast_t* type, ast_t* scope)
 
       typeparam_current_inner(left, scope);
       typeparam_current_inner(right, scope);
+
+      break;
     }
 
     default:

